.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        node: [12.x, 14.x, 15.x]
+      fail-fast: false
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set node version to ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Versions
+        run: yarn versions
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      # - name: Check outdated dependencies
+      #   if: github.ref == 'refs/heads/main'
+      #   run: yarn outdated
+
+      - name: Build vite
+        run: yarn build-vite
+
+      - name: Build plugin-vue
+        run: yarn build-plugin-vue
+
+      - name: Lint
+        if: matrix.os == 'ubuntu-latest' && matrix.node == '14.x'
+        run: yarn lint
+
+      - name: Test serve
+        run: yarn test-serve --runInBand
+
+      - name: Test build
+        run: yarn test-build --runInBand
+
+      # - name: Audit dependencies
+      #   run: yarn audit --groups dependencies
+
+      # - name: Audit peerDependencies
+      #   run: yarn audit --groups peerDependencies
```

Comment:
It's arguable if we need audit this way
I think dependabot may work for a repo like vite